azurerm/resource_arm_container_group.go

```diff
@@ -0,0 +1,294 @@
+
+		// 	for _, envVarRaw := range envVars {
+		// 		envVar := containerinstance.EnvironmentVariable{
+		// 			Name:  &envVarRaw.get("name").(string),
+		// 			Value: &envVarRaw.get("value").(string),
+		// 		}
+		// 		envVarsList = append(envVarsList, envVar)
+		// 	}
+		// 	container.EnvironmentVariables = &envVarsList
+		// }
+
+		containers = append(containers, container)
+	}
+
+	containerGroup := containerinstance.ContainerGroup{
+		Name:     &name,
+		Location: &location,
+		Tags:     expandTags(tags),
+		ContainerGroupProperties: &containerinstance.ContainerGroupProperties{
+			Containers: &containers,
+			IPAddress: &containerinstance.IPAddress{
+				Type:  &IPAddressType,
+				Ports: &containerGroupPorts,
+			},
+			OsType: containerinstance.OperatingSystemTypes(OSType),
+		},
+	}
+
+	_, error := containerGroupsClient.CreateOrUpdate(resGroup, name, containerGroup)
+	if error != nil {
+		return error
+	}
+
+	read, readErr := containerGroupsClient.Get(resGroup, name)
+	if readErr != nil {
+		return readErr
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read container group %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmContainerGroupRead(d, meta)
+}
+func resourceArmContainerGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["containerGroups"]
+
+	resp, error := containterGroupsClient.Get(resGroup, name)
+
+	if error != nil {
+		return error
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	flattenAndSetTags(d, resp.Tags)
+
+	d.Set("os_type", string(resp.OsType))
+	d.Set("ip_address_type", *resp.IPAddress.Type)
+	d.Set("ip_address", *resp.IPAddress.IP)
+
+	containers := *resp.Containers
+
+	containerConfigs := make([]interface{}, 0, len(containers))
+	for _, container := range containers {
+		containerConfig := make(map[string]interface{})
+		containerConfig["name"] = *container.Name
+		containerConfig["image"] = *container.Image
+
+		resourceRequests := *(*container.Resources).Requests
+		containerConfig["cpu"] = *resourceRequests.CPU
+		containerConfig["memory"] = *resourceRequests.MemoryInGB
+
+		if len(*container.Ports) > 0 {
+			containerConfig["port"] = *(*container.Ports)[0].Port
+		}
+		// protocol isn't returned in container config
+
+		containerConfigs = append(containerConfigs, containerConfig)
+	}
+
+	d.Set("container", containerConfigs)
+
+	return nil
+}
+func resourceArmContainerGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	// container group properties
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
```

Comment:
rather than using the state - can we parse these from the URL as in the Get method? i.e.

```
id, err := parseAzureResourceID(d.Id())
if err != nil {
  return err
}

resGroup := id.ResourceGroup
name := id.Path["containerGroups"]
```

Reply:
Done.